### ISSUE TITLE
jenkins2-ha-agents does not scale down if there is at least one job running

### ISSUE THREAD

**Issue by Ten0** [widdix/aws-cf-templates#677]
On this template:
https://github.com/widdix/aws-cf-templates/blob/e80083b4aa336ed8882d65d7c359a271e24c65e8/jenkins/jenkins2-ha-agents.yaml#L2326-L2332

The alarm only triggers if there are absolutely no jobs left running.
This means that if e.g. during the day you need on average 20 agents, max 50, but never less than 1, you'll still always pay for 50 agents because as you always have one job running the alarm never triggers.

It looks like this could be fixed by:
 - Making this alarm trigger if there are agents that don't do anything
 - Using scale-in protection (on ASG) to prevent instances that currently have jobs running from being picked for termination

**Comment by michaelwittig** [comment]
I agree. The tricky part seems to set scale in protection in a reliable fashion as quickly as possible after a job runs on an agent.
